### PR TITLE
[DI] add tags `container.preload`/`.no_preload` to declare extra classes to preload/services to not preload

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

To allow fine-grained declaration of sidekick classes in DI extensions.
Follows #36103

### CODE REVIEW

**Comment by michaljusiega** [comment]
So `class_exists` is no longer needed in favor of `container.preload` tag do add classes to `*.preload.php` dumped file ?

**Comment by stof** [comment]
@michaljusiega AFAIU, the class_exists were solving 2 issues:

- discovery of classes to be preloaded
- compatibility with preloading is some specific cases related to anonymous classes which break if the class is not loaded

This PR provides an alternative solution to the first issue (though `class_exists` will still work). For the second problem, `class_exists` is still the proper solution IMO, as that's not about the Symfony preload dumper by itself, but about compat with preloading.

**Comment by nicolas-grekas** [comment]
I updated the PR to now also provide a `container.no_preload` tag.
I added it to eg. console commands.

@michaljusiega: what @stof said: `class_exists()` are better options for portability and for locality. `container.preload` serves adding preloading info to third party packages that don't (want to) use `class_exists()`, and also in programmatic situation where a bundle's logic might known better about fine grained preloading, depending on the configuration.

**Comment by michaljusiega** [comment]
@nicolas-grekas, @stof I understand. Thanks!

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -78,6 +78,7 @@ class PhpDumper extends Dumper
     private $namespace;
     private $asFiles;
     private $hotPathTag;
+    private $preloadTag;
```
Comment (rosier) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]:
```suggestion
    private $preloadTags;
```

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**